examples/demo-app/src/cloud-providers/aws/aws-provider.js

```diff
@@ -0,0 +1,291 @@
+
+  async _getMapListFromStorage(level) {
+    let mapList = [];
+    let capitalizedLevel = level.charAt(0).toUpperCase() + level.slice(1);
+    await Storage.list('', {level})
+      .then((result) => {
+        mapList = AwsProvider._updateFileList(result, level);
+      })
+      .catch((e) => {
+        this._handleError(`${capitalizedLevel} map load failed`, e);
+      });
+    return mapList;
+  }
+
+  static async _updateFileList(fileList, level) {
+    let updatedFileList = [];
+
+    for (const file of fileList) {
+      if (!file.key.endsWith('.json')) {
+        continue;
+      }
+      const title = file.key.split('.')[0];
+      const thumbnailKey = title + '.png';
+      const thumbnailURL = await Storage.get(thumbnailKey, {
+        level: level,
+        download: false
+      });
+
+      const description = await Storage.get(thumbnailKey, {
+        level: level,
+        download: true
+      }).then((thumbnail) =>
+        thumbnail.Metadata.desc ? atob(thumbnail.Metadata.desc) : 'No description available.'
+      );
+      const mapIsPrivate = level === 'private';
+
+      updatedFileList.push({
+        id: file.key,
+        title: title,
+        description: description,
+        privateMap: mapIsPrivate,
+        thumbnail: thumbnailURL,
+        lastModification: new Date(Date.parse(file.lastModified)),
+        loadParams: {
+          mapId: file.key,
+          privateMap: mapIsPrivate,
+          level: level
+        }
+      });
+    }
+    return updatedFileList;
+  }
+
+  /**
+   * Required!
+   *
+   * @returns {MapResponse}
+   */
+  async downloadMap(loadParams) {
+    const {level, mapId, identityId} = loadParams;
+    let mapURL =
+      level === 'private'
+        ? await Storage.get(mapId, {level})
+        : await Storage.get(mapId, {level, identityId});
+
+    const mapData = await fetch(mapURL).then((response) => response.json());
+    this._loadParam = loadParams;
+
+    return {
+      map: mapData,
+      format: 'keplergl'
+    };
+  }
+
+  async uploadMap({mapData, options = {}}) {
+    const {isPublic, overwrite} = options;
+    const {map, thumbnail} = mapData;
+    const {title, description} = map && map.info;
+    const name = title;
+    // Since we share through a map link, this could be private as well
+    const level = isPublic ? 'protected' : 'private';
+    let mapId = '';
+    let identityId = '';
+
+    await Auth.currentUserInfo().then((userInfo) => {
+      identityId = userInfo.id;
+    });
+    await Storage.put(name + '.png', thumbnail, {
+      level,
+      contentType: 'images/png',
+      metadata: {desc: btoa(description)}
+    }).catch((e) => {
+      this._handleError('Saving failed', e);
+    });
+
+    await Storage.put(name + '.json', map, {
+      level,
+      contentType: 'application/json'
+    })
+      .then((response) => {
+        mapId = response && response.key;
+      })
+      .catch((e) => {
+        this._handleError('Saving failed', e);
+      });
```

Comment:
AWS MetaData only supports US-Ascii. If we want to support e.g. UTF-8 as well, we need to find another way to save the description. For example in a ${mapName}.txt.

Reply:
Yes we need to fix that or convert name and description to US ASCII before saving. I can't access my AWS storage anymore after storing a map with name "Münchener Straße" and same description

Reply:
I think the latter problem exists because we don't use a proper encoded URI when requesting the map and review in listMaps. The URI according to AWS S3 console ends with `UTF8-TEST+Name+M%C3%BCnchener+Stra%C3%9Fe+%40+Reply.de.png` but the URI requested ends with `UTF8-TEST%20Name%20M%C3%BCnchener%20Stra%C3%9Fe%20%40%20Reply.png` . The name and description I provided was `UTF8-TEST Name Münchener Straße @ Reply.de`

Reply:
Done